tests/fortran/prune_unused_children_test.py

```diff
@@ -0,0 +1,776 @@
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
+from typing import Dict, List
+
+from fparser.two.Fortran2003 import Program
+from fparser.two.parser import ParserFactory
+from fparser.two.utils import walk
+
+from dace.frontend.fortran.ast_desugaring import ENTRY_POINT_OBJECT_CLASSES, NAMED_STMTS_OF_INTEREST_CLASSES, \
+    find_name_of_node, prune_unused_objects, SPEC, ident_spec
+from dace.frontend.fortran.ast_utils import children_of_type, singular
+from dace.frontend.fortran.fortran_parser import construct_full_ast
+from tests.fortran.fortran_test_helper import SourceCodeBuilder
+
+
+def parse_and_improve(sources: Dict[str, str]):
```

Comment:
This function appears in many places, please refactor to `fortran_test_helper` (as well as other relevant functions)